realm/realm-library/src/main/cpp/java_binding_context.cpp

```diff
@@ -17,18 +17,40 @@
 #include "java_binding_context.hpp"
 #include "jni_util/java_method.hpp"
 
+#include "util.hpp"
+
 using namespace realm;
 using namespace realm::_impl;
 using namespace realm::jni_util;
 
-void JavaBindingContext::changes_available()
+void JavaBindingContext::before_notify()
 {
+    if (JniUtils::get_env()->ExceptionCheck()) {
+        return;
```

Comment:
Case for `SyncClient.run()`, yes, we should throw C++ exception and use `CATCH_STD` to convert, since in that case, `run` is first call when java enter c++ land.

But here is totally different, the first JNI call is the ALooper's callback which is inside the object store. We cannot do `CATCH_STD` there.